src/wp-includes/media.php

```diff
@@ -3751,6 +3751,12 @@ function wp_get_image_editor( $path, $args = array() ) {
 		}
 	}
 
+	/** This filter is documented in wp-includes/class-wp-image-editor.php */
+	$output_format = apply_filters( 'image_editor_output_format', array(), $path, $args['mime_type'] );
+	if ( isset( $output_format[ $args['mime_type'] ] ) ) {
+		$args['mime_type'] = $output_format[ $mime_type ];
```

Comment:
Thanks so much for the patch!
If I'm reading this correctly, it looks like this replaces which mime type is checked, rather than checking both, correct?

Assuming so, I think it'd be a good idea to check both. It's an edge case, but for instance, if someone uploads a WebP, and the filter is configured to always output JPEG, this should not select an image editor that doesn't support loading WebP.

Reply:
I've wondered if it might be possible to support sending an array of mime types for checking this.

Reply:
good suggestion, I'll try to update this so it checks both.

Reply:
I updated the implementation to set a "output_mime_type" when mapped, then in _wp_image_editor_choose looking for an implementation that supports both input and output types.